consensus/istanbul/core/prepare_test.go

```diff
@@ -81,3 +86,86 @@ func TestCore_sendPrepare(t *testing.T) {
 		mockCtrl.Finish()
 	}
 }
+
+func BenchmarkMsgCmp(b *testing.B) {
+	getEmptySubject := func() istanbul.Subject {
+		return istanbul.Subject{
+			View: &istanbul.View{
+				Round:    big.NewInt(0),
+				Sequence: big.NewInt(0),
+			},
+			Digest:   common.HexToHash("1"),
+			PrevHash: common.HexToHash("2"),
+		}
+	}
+	s1, s2 := getEmptySubject(), getEmptySubject()
+
+	// Worst
+	b.Run("reflect.DeepEqual", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			reflect.DeepEqual(s1, s2)
+		}
+	})
+
+	// Better
+	b.Run("EqualImpl", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			s1.Equal(&s2)
+		}
+	})
+}
+
+func TestSubjectCmp(t *testing.T) {
```

Comment:
Perhaps we can use `reflect.DeepEqual` to prove that `reflect.DeepEqual(s1,s2) is equivalent to s1.Equal(s2)` 

Since this PR replaces `reflect.DeepEqual` with `Subject.Equal`. I think it's natural to "verify this improvement does not make any problems with consensus" by comparing the two functions.

Reply:
Agree on half, concerns on another half. If that code added, it remains debris and the following readers may confuse why this unnecessary comparison is placed. For now, I pushed new one for your comment